ddtrace/contrib/celery/__init__.py

```diff
@@ -12,36 +12,33 @@
     def my_task():
         pass
 
-
     class MyTask(app.Task):
         def run(self):
             pass
 
 
-If you don't need to patch all Celery tasks, you can patch individual
-applications or tasks using a fine grain patching method::
+To change Celery service name, you can update the attached ``Pin``
+instance::
 
-    import celery
-    from ddtrace.contrib.celery import patch_app, patch_task
+    from ddtrace import Pin
 
-    # patch only this application
     app = celery.Celery()
-    app = patch_app(app)
 
-    # or if you didn't patch the whole application, just patch
-    # a single function or class based Task
     @app.task
-    def fn_task():
+    def compute_stats():
         pass
 
+    # globally
+    Pin.override(app, service='background-jobs')
+
+    # by task
+    Pin.override(compute_stats, service='data-processing')
 
-    class BaseClassTask(celery.Task):
-        def run(self):
-            pass
 
+By default, reported service names are:
+    * ``celery-producer`` when messages are sent for processing
```

Comment:
Just a question: is `message` the Celery vocabulary for a task instance?

Reply:
I guess it's an implementation detail. Better to make it more generic.